extensions/infinispan-cache/runtime/src/main/java/io/quarkus/cache/infinispan/runtime/InfinispanCacheImpl.java

```diff
@@ -81,16 +86,11 @@ private <T> T decodeNull(Object value) {
 
     @Override
     public <K, V> Uni<V> get(K key, Function<K, V> valueLoader) {
-        return Uni.createFrom()
-                .completionStage(() -> CompletionStages.handleAndCompose(remoteCache.getAsync(key), (v1, ex1) -> {
-                    if (ex1 != null) {
-                        return CompletableFuture.failedFuture(ex1);
-                    }
-
+        CompletableFuture<V> infinispanGet = threadContext.withContextCapture(remoteCache.getAsync(key))
```

Comment:
We need a supplier, as that code should run when the uni is subscribed to.

So, a supplier of completion stage would be more correct.